priv/perf/apps/omg_load_testing/lib/utils/faucet.ex

```diff
@@ -0,0 +1,136 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.LoadTesting.Utils.Faucet do
+  @moduledoc false
+
+  require Logger
+
+  use GenServer
+
+  alias ExPlasma.Encoding
+  alias ExPlasma.Utxo
+  alias OMG.LoadTesting.Utils.Account
+  alias OMG.LoadTesting.Utils.ChildChain
+  alias OMG.LoadTesting.Utils.Deposit
+  alias OMG.LoadTesting.Utils.Ethereum, as: Eth
+
+  @eth <<0::160>>
+  @fund_child_chain_account_retries 100
+
+  defstruct [:account, :fee, utxos: %{}]
+
+  def start_link() do
+    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  end
+
+  def fund_child_chain_account(account, amount, token) do
+    GenServer.call(__MODULE__, {:fund_child_chain, account, amount, token}, :infinity)
+  end
+
+  def get_faucet() do
+    GenServer.call(__MODULE__, :get_faucet, :infinity)
+  end
+
+  def init(_) do
+    opts = fetch_default_opts()
+    fee_wei = Keyword.fetch!(opts, :fee_wei)
+
+    {faucet, eth_utxo} = get_funded_faucet_account(opts)
+    Logger.debug("Using faucet: #{Encoding.to_hex(faucet.addr)}")
+
+    state = struct!(__MODULE__, account: faucet, fee: fee_wei, utxos: %{@eth => eth_utxo})
+    {:ok, state}
+  end
+
+  def handle_call(:get_faucet, _from, %__MODULE__{account: faucet} = state) do
+    {:reply, {:ok, faucet}, state}
+  end
+
+  def handle_call(
+        {:fund_child_chain, account, amount, @eth = token},
+        _from,
+        %__MODULE__{account: faucet} = state
+      ) do
+    {utxo, utxo_amount} = state.utxos[token]
+    change = utxo_amount - amount - state.fee
+    if change < 0, do: raise({:error, :change_below_zero})
```

Comment:
this would kill the process... any reason why you're doing this?

Reply:
the test can never successfully run with the setup.....idk, in this case shall we still return `{:error}` instead of raise to stop the test early?

Reply:
then there needs to be a hierarchical dependency between processes that would tear the application (test) down. this seems a bit random

Reply:
oh I see, so sth like calling Genserver `terminate`?

Reply:
it would mean that you create a hierarchical dependency between processes:
```
application -> supervisor -> start_link faucet 
                                             -> start other children
```
so that if faucet process crashes, it tears down the application (because exceptions are prograded upwards through bidirectional process links)

Reply:
One other approach could be making another deposit in case of insufficient funds - caller would have to wait longer in some case but the application would not crash.